src/components/layout/breadcrumb/index.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react'
+import * as S from './styles'
+import Link from '../../links/link'
+import { ButtonSize } from '../../buttons'
+
+export interface Crumb {
+  label: string
+  path?: string
+}
+
+export interface BreadcrumbProps {
+  crumbs?: Crumb[]
+  homeLabel?: string
+  homeLink?: string
+}
+
+const Breadcrumb: React.FC<BreadcrumbProps> = ({
+  crumbs,
+  homeLabel = 'Homepage',
+  homeLink = '/',
+}: BreadcrumbProps) => {
+  return (
+    <S.OrderedList>
+      {crumbs && (
+        <S.ListItem>
+          <Link size={ButtonSize.Small} to={homeLink}>
+            {homeLabel}
+          </Link>
+        </S.ListItem>
+      )}
+
+      {crumbs?.map((c, i) => {
+        return (
+          <>
+            <S.Separator>/</S.Separator>
+            <S.ListItem>
+              {i == crumbs.length - 1 || c.path == null ? (
```

Comment:
Tady doporučuji aplikovat trochu více funkcionální přístup. 

Konkrétně transformovat `crumbs` na poslední a zbylé: 
```typescript
function transformCrumbs(
  crumbs: Crumb[]
): { lastCrumb: Crumb; firstCrumbs: Crumb[] } {
  // ... immutable implementation (you can use .pop but over a copy)
}
```

a pak vrátit JSX bez podmínek a iterací: 
```typescript jsx
return (
    <S.OrderedList>
      {crumbs && (
        <S.ListItem>
          <Link size={ButtonSize.Small} to={homeLink}>
            {homeLabel}
          </Link>
        </S.ListItem>
      )}

      {firstCrumbs?.map((crumb) => (
        <>
          <S.Separator>/</S.Separator>
          <S.ListItem>
            <Link size={ButtonSize.Small} to={crumb.path}>
              {crumb.label}
            </Link>
          </S.ListItem>
        </>
      ))}
      <S.Separator>/</S.Separator>
      <S.ListItem>
        <S.Current>{lastCrumb.label}</S.Current>
      </S.ListItem>
    </S.OrderedList>
  )
```

Reply:
Tohle mi z toho vylezlo, aby byly ošetřené všechny případy (např. když někdo zapomene vyplnit path u prostředního drobku, pole je krátké atp...
![image](https://user-images.githubusercontent.com/1295186/104107815-be4ed280-52bf-11eb-831b-1d2903714bfb.png)

Celé je to dvakrát delší než původně a čitelnost se zhoršila. Mě osobně se líbí víc ta moje původní varianta. Ale nechávám na tobě, házím to ve zvláštním commitu, takže lze použít i nepoužít.

Reply:
@maral, neobdržel jsem na Slacku odpověď takže jsem to za tebe dotáhl. Můžes se podívat na pushnuté změny pro inspiraci. Pokud se validace provedou na začátku, množství podmínek se zkrátí a tím pádem i layout. Ještě jsem pro jistotu doplnil test pro dvojí kontrolu případných změn do budoucna. 🙂

Reply:
Jo jo, to je ta moje nezkušenost. Vypadá to dobře, díky za dotažení ;)